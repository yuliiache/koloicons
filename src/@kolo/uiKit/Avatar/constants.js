export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
