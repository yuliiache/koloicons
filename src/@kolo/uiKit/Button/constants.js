export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
