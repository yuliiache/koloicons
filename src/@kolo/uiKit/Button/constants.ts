export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
