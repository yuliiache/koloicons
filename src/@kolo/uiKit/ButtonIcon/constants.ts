export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
