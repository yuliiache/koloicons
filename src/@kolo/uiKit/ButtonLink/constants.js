export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

