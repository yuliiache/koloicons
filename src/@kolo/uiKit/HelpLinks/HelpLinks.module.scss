@import "@kolo/design";

.helpLinkList {
  display: flex;
  gap: 16px;

  @media (max-width: $mobileWidth) {
    gap: 32px;
  }
}
