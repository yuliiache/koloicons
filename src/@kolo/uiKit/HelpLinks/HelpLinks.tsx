import {URLS} from '@kolo/constants/constants';
import {FC} from 'react';

import Link from '../Link/Link';
import styles from './HelpLinks.module.scss';

interface HelpLink {
  name: string;
  url: string;
}

export const HELP_LINKS: readonly HelpLink[] = [
  {name: 'Privacy Policy', url: URLS.privacyPolicy},
  {name: 'Terms of Service', url: URLS.termsOfService},
  {name: 'Cookie Policy', url: URLS.cookiePolicy},
];

const HelpLinks: FC = () => {
  return (
    <ul className={styles.helpLinkList}>
      {HELP_LINKS.map(({name, url}: HelpLink) => (
        <li key={name}>
          <Link
            url={url}
            isInNewTab
          >
            {name}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default HelpLinks;
