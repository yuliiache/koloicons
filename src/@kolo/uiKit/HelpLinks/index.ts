import HelpLinks from './HelpLinks';

export default HelpLinks;
