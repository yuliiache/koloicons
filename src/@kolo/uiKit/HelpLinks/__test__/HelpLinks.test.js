import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {INTERNAL_LINK_TEST_ID} from '@kolo/uiKit/Link/constants';
import {cleanup, render, screen} from '@testing-library/react';

import {HELP_LINKS} from '../HelpLinks';
import HelpLinks from '../HelpLinks';

describe('HelpLinks component', () => {
  afterEach(cleanup);

  it('should successfully render snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <HelpLinks />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should successfully render with correct items', () => {
    render(
      <MemoryRouter>
        <HelpLinks />
      </MemoryRouter>
    );

    HELP_LINKS.forEach(({name, url}) => {
      const helpLink = screen.getByText(name);
      expect(helpLink).toBeInTheDocument();
      expect(helpLink.getAttribute('href')).toBe(url);
    });

    const helpLinks = screen.getAllByTestId(INTERNAL_LINK_TEST_ID);

    expect(helpLinks).toHaveLength(3);
  });
});
