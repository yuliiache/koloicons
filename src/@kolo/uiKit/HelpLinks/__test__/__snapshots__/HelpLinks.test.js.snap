// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpLinks component should successfully render snapshot 1`] = `
<DocumentFragment>
  <ul
    class="helpLinkList"
  >
    <li>
      <a
        class="link link--default"
        data-testid="internalLinkComponent"
        href="/privacy-policy-docs"
        rel="noopener noreferrer"
        target="_blank"
      >
        Privacy Policy
      </a>
    </li>
    <li>
      <a
        class="link link--default"
        data-testid="internalLinkComponent"
        href="/terms-of-service"
        rel="noopener noreferrer"
        target="_blank"
      >
        Terms of Service
      </a>
    </li>
    <li>
      <a
        class="link link--default"
        data-testid="internalLinkComponent"
        href="/cookie-policy-docs"
        rel="noopener noreferrer"
        target="_blank"
      >
        Cookie Policy
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
